utils/fswriters.go

```diff
@@ -0,0 +1,71 @@
+package utils
+
+import (
+	"io"
+	"os"
+	"path/filepath"
+)
+
+type atomicFileWriter struct {
+	f        *os.File
+	path     string
+	writeErr error
+	perm     os.FileMode
+}
+
+func (a *atomicFileWriter) Write(p []byte) (n int, err error) {
+	n, err = a.f.Write(p)
+	if err != nil {
+		a.writeErr = err
+	}
+	return n, err
+}
+
+func (a *atomicFileWriter) Close() (err error) {
```

Comment:
close will commit the file, right? how about "close", I guess Close is not recommended using by caller